### PR TITLE
Add parameter `about_point` for `rotate`

### PR DESCRIPTION
**Pull request by kolibril13**
## Changelog / Overview

 `about_point` parameter for `rotate`

## Motivation

As `about_point = ...` and `rotate` are often used together, after this pr  this script will have IDE autocompletion:
```python
class MovingLine(Scene):
    def construct(self):
        line1 = Line(ORIGIN, 2*RIGHT)
        line2 = Line(ORIGIN, 2*RIGHT).rotate(80*DEGREES, about_point=2*LEFT)
        line3 = Line(RIGHT, ORIGIN).rotate(20*DEGREES, about_point=UP)
        self.add(line3)
        self.add(line1, line2,line3)
        self.wait()
```
## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Review by jsonvillanueva — COMMENTED** [review]
I think this should also be done for the opengl_mobject.py rotate

**Comment by kolibril13** [comment]
> I think this should also be done for the opengl_mobject.py rotate

Good idea.
The tests are failing and I can't figure out how to solve this:
```py
    def rotate_about_number(self, number, angle, axis=OUT, **kwargs):
>       return self.rotate(angle, axis, about_point=self.n2p(number), **kwargs)
E       TypeError: rotate() got multiple values for argument 'about_point'
```
If someone has an idea, feel free to commit on this branch.

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -1021,7 +1021,13 @@ def rotate_about_origin(self, angle, axis=OUT, axes=[]):
         """Rotates the :class:`~.Mobject` about the ORIGIN, which is at [0,0,0]."""
         return self.rotate(angle, axis, about_point=ORIGIN)
 
-    def rotate(self, angle, axis=OUT, **kwargs):
+    def rotate(
+        self,
+        angle,
+        about_point: Union[np.ndarray, List, None] = None,
```
Comment (abhi3019) [inline manim/mobject/mobject.py:1027]: Switching the order of these 2 parameters should fix the issue @kolibril13

**Comment by abhi3019** [comment]
Switching the order of [these 2 parameters](https://github.com/ManimCommunity/manim/pull/1238/files/50d7685b9c80ac87e041b1e4981c8a1b6e8e5c22#diff-fa8fe466fb7275e64543789145cbd2e35b267715610a6376591ddbda0d2f6194R1027-R1028) should fix the issue @kolibril13

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -1021,7 +1021,13 @@ def rotate_about_origin(self, angle, axis=OUT, axes=[]):
...
+        axis=OUT,
```
Comment (kolibril13) [inline manim/mobject/mobject.py:1028]:
```suggestion
        about_point: Union[np.ndarray, List, None] = None,
        axis=OUT,
```
```suggestion
        axis=OUT,
        about_point: Union[np.ndarray, List, None] = None
```

**Comment by kolibril13** [comment]
> Switching the order of [these 2 parameters](https://github.com/ManimCommunity/manim/pull/1238/files/50d7685b9c80ac87e041b1e4981c8a1b6e8e5c22#diff-fa8fe466fb7275e64543789145cbd2e35b267715610a6376591ddbda0d2f6194R1027-R1028) should fix the issue @kolibril13

Thanks, @abhi3019

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -1021,7 +1021,13 @@ def rotate_about_origin(self, angle, axis=OUT, axes=[]):
         """Rotates the :class:`~.Mobject` about the ORIGIN, which is at [0,0,0]."""
         return self.rotate(angle, axis, about_point=ORIGIN)
 
-    def rotate(self, angle, axis=OUT, **kwargs):
+    def rotate(
+        self,
+        angle,
+        axis=OUT,
+        about_point: Union[np.ndarray, List, None] = None
```
Comment (abhi3019) [inline manim/mobject/mobject.py:1028]: missing a comma here
```suggestion
        about_point: Union[np.ndarray, List, None] = None,
```

File: manim/mobject/opengl_mobject.py [hunk 4]
```diff
@@ -614,7 +615,13 @@ def func(points):
     def rotate_about_origin(self, angle, axis=OUT):
         return self.rotate(angle, axis, about_point=ORIGIN)
 
-    def rotate(self, angle, axis=OUT, **kwargs):
+    def rotate(
+        self,
+        angle,
+        axis=OUT,
+        about_point: Union[np.ndarray, List, None] = None,
```
Comment (kolibril13) [inline manim/mobject/opengl_mobject.py:622]: I've now added this parameter to opengl as well, but I have  never worked on it and don't know how to test it /if there are any tests. @eulertour : can you confirm that this is correct?

Comment (eulertour) [reply manim/mobject/opengl_mobject.py:622]: It won't break anything if that's your concern

**Review by eulertour — APPROVED** [review]

File: manim/mobject/opengl_mobject.py (hunk 4 above)
Comment (kolibril13) [reply manim/mobject/opengl_mobject.py:622]: Okay, perfect, yes that was the concern.

**Pull request merged**